### ISSUE TITLE
[v1.0.7-2] ES Module is not properly exported

### ISSUE THREAD

**Issue by Vinnl** [linkeddata/rdflib.js#381]
In `package.json`, the `module` key refers to `src/index.js` - which no longer exists, because it's been converted to TypeScript.

Typically, this should refer to a non-bundled version of the code, and you can presume many recent ES features to be available. In other words, you want this to refer to a file that only has the type definitions stripped. (See for instance [how Tripledoc also exports a version with ES module support to the file defined in the `<package.json>.module`](https://gitlab.com/vincenttunru/tripledoc/blob/master/rollup.config.js#L11).)

If that's too much work, it's probably a good idea to remove the `module` key for now, so bundles that support ES modules will fallback to using the bundled version.

**Comment by theWebalyst** [comment]
FYI here's an example on the rdflib chat where this breaks. If you follow the link to the live REPL it generates an error after a few seconds when the build fails to find `lib/index.js`:

https://gitter.im/linkeddata/rdflib.js?at=5e0e552d04a1900497f2af60

**Comment by Vinnl** [comment]
Note that it's not `lib/index.js` that cannot be found (which is referenced by `main` in `package.json`), but `src/index.js` (referenced by `module`):

![image](https://user-images.githubusercontent.com/4251/71722361-d415f600-2e28-11ea-9465-eaac3181c134.png)

(Seen in [the REPL](https://svelte.dev/repl/65165231684f4c28abdb06bf87a130d5?version=3.16.7).)

**Comment by SirkoS** [comment]
I might be missing something, but isn't reestablishing ESM-support just a few lines in the configs?

I setup a small fork for testing at [SirkoS/rdflib.js](https://github.com/SirkoS/rdflib.js) and it seems to work just fine.

If you agree, I can just create a pull request.

**Comment by Vinnl** [comment]
@SirkoS Yes, I'd certainly expect it to look something like [that](https://github.com/SirkoS/rdflib.js/commit/4a3dc1419e42fd937195b9faeb382c0915ec34f7), so I think a PR would be a good idea.

**Comment by jeswr** [comment]
Closing as resolved: in v2.4.0 the package.json module field points to esm/index.js, and the published npm tarball contains that file, so the original problem (module referencing a nonexistent src/index.js) is gone. Remaining problems with actually importing the esm build are tracked in #657. Verified as part of the maintainer-signed-off triage (plan #824); please reopen with a fresh repro if the module entry itself is broken again.
